### PR TITLE
replace distribution chart with better node stats

### PR DESCRIPTION
**Pull request by olizilla**
wip on #887 to dial down geolocation activity

webui-connection-status

screenshot 2018-12-10 at 22 50 27

screenshot 2018-12-11 at 09 01 38

License: MIT
Signed-off-by: Oli Evans <oli@tableflip.io>

### CODE REVIEW

**Comment by olizilla** [comment]
@lidel this PR comes from 

1. A need to only do gelocation look-ups on the peers page, which means we can't have the distribution chart on the status page, as we'd have no data for it.
2. Usability testing, that showed it wasn't explicit from the status page that you were connected to IPFS, the need for some context for new users, hence the "connected to IPFS, Sharing nMB, discovered x peers.
3. Refactoring the status widgets into smaller pieces so I can move them around and experiement.

The code needs tidying up, and i18n, but It'd help to get a quick thumbs up from y'all (@lidel @fsdiogo  @hacdias) or any questions concerns you might have.

Of note, this doesn't give us the full repo size widget that @lidel came up with in https://github.com/ipfs-shipyard/ipfs-webui/issues/887#issuecomment-440330288 but that will need some design effort, so we can tackle it as a seperate PR.

**Comment by olizilla** [comment]
Some cool things I'd like to draw attention too

- Improved html semantics [Box is now a `section` element](https://github.com/ipfs-shipyard/ipfs-webui/pull/902/files#diff-974895d3f64e98d54f539c8e3eab26feR11)
- We have a sematically cool [Definition component](https://github.com/ipfs-shipyard/ipfs-webui/pull/902/files#diff-af08ee7b418a7728ae29a5bdb1005462) to make styling of key/value definition lists reusable and using the right dl/dt/dd elements under the hood. Here is usage example:

https://github.com/ipfs-shipyard/ipfs-webui/blob/92823a451679e72fbf4822c92b0d65a6f567014b/src/status/NodeInfo.js#L29-L32

- A Details component to for collapsable content that just serves to make sure the `details` element polyfill is  included for [Edge](https://caniuse.com/#feat=details)
- A super fun, [Shell component](url) for when you really want to show a terminal output example.

https://github.com/ipfs-shipyard/ipfs-webui/blob/92823a451679e72fbf4822c92b0d65a6f567014b/src/status/StatusNotConnected.js#L16-L20

File: src/status/NetworkTraffic.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      <div>
+        <Title>{t('networkTraffic')}</Title>
+        <div className='flex flex-column justify-between' style={{ maxWidth: 400 }}>
+          <div className='mr2 ml2 mt3 mt0-l'>
+            <Speedometer
+              title={t('upSpeed')}
+              color='#f39021'
+              {...upSpeed} />
+          </div>
+          <div className='mr2 ml2 mt3 mt0-l'>
```
Comment (fsdiogo) [inline src/status/NetworkTraffic.js]:
```suggestion
          <div className='mh2 mt3 mt0-l'>
```

File: src/status/NetworkTraffic.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  render () {
+    const { t } = this.props
+    const { downSpeed, upSpeed } = this.state
+
+    return (
+      <div>
+        <Title>{t('networkTraffic')}</Title>
+        <div className='flex flex-column justify-between' style={{ maxWidth: 400 }}>
+          <div className='mr2 ml2 mt3 mt0-l'>
```
Comment (fsdiogo) [inline src/status/NetworkTraffic.js]:
```suggestion
          <div className='mh2 mv3 mt0-l'>
```

`mv3` to give it a bit more breathing room.

File: src/status/StatusPage.js [hunk 3]
```diff
@@ -1,28 +1,53 @@
...
         </div>
-        <div className='w-100 w-40-l pl0 pl2-l flex-none'>
-          <CountryChart />
+      </Box>
+      <Box className='mt3 pa3' style={{ opacity: ipfsConnected ? 1 : 0.4 }}>
+        <div className='flex flex-column flex-row-l'>
+          <div className='pr0 pr2-l flex-auto'>
+            <NodeBandwidthChart />
+          </div>
+          <div className='dn db-l pl3 pr5'>
```
Comment (fsdiogo) [inline src/status/StatusPage.js]: Why hide the speedometers in narrow viewports? We can just put them below the chart.

Comment (olizilla) [reply src/status/StatusPage.js]: They look kinda odd dangling below the graph.

Comment (fsdiogo) [reply src/status/StatusPage.js]: Alright, we can take care of that on another PR.

**Pull request merged**